### PR TITLE
[webui] Allow editing of version for Kiwi files.

### PR DESCRIPTION
**Pull request by evanrolfe**

### CODE REVIEW

File: src/api/app/models/kiwi/image.rb [hunk 1]
```diff
@@ -37,9 +37,8 @@ class Image < ApplicationRecord
     validate :check_use_project_repositories
     validate :check_package_groups
     validates :preference, presence: true
-    accepts_nested_attributes_for :preference, reject_if: proc { |attributes|
-      attributes['type_containerconfig_name'].blank? && attributes['type_containerconfig_tag'].blank?
```
Comment (Ana06) [inline src/api/app/models/kiwi/image.rb]: who is that related to the version? 🤔

**Comment by Ana06** [comment]
@evanrolfe rebase! 😉

File: src/api/app/models/kiwi/image.rb (hunk 1 above)
Comment (ChrisBr) [reply src/api/app/models/kiwi/image.rb]: Because before we only had containerconfiguration in the preference model and we rejected the nested attributes if it didn't contain a containerconfig name or tag. Now we also have a version in the preference so this is not valid anymore.

File: src/api/app/models/kiwi/image/xml_builder.rb [hunk 2]
```diff
@@ -44,6 +46,18 @@ def update_preferences(document)
         document
       end
 
+      def update_preference_version(document)
+        if @image.preference.version.present?
+          if document.xpath('image/preferences/version').any?
+            document.xpath('image/preferences/version').remove
```
Comment (ChrisBr) [inline src/api/app/models/kiwi/image/xml_builder.rb]: I would prefer to replace and not remove to keep the diff as small as possible. So you would need two different cases.

File: src/api/app/models/kiwi/image.rb (hunk 1 above)
Comment (evanrolfe) [reply src/api/app/models/kiwi/image.rb]: because the preferences xml is like this:
```xml
  <preferences>
    <type image="docker" boot="grub">
      <containerconfig
        name="my_container"
        tag="latest" />
        <oemconfig>test</oemconfig>
    </type>
    <bootsplash-theme>gnome</bootsplash-theme>
    <bootloader-theme>gnome-dark</bootloader-theme>
    <version>2.0</version>
  </preferences>
```

And before we only stored the containerconfig name & tag so it didn't make sense to save preferences if there were no containerconfig attributes set. Now thats different since we also store preferences.version.

File: src/api/app/models/kiwi/image/xml_builder.rb (hunk 2 above)
Comment (evanrolfe) [reply src/api/app/models/kiwi/image/xml_builder.rb]: @ChrisBr what do you mean "to keep the diff as small as possible"? The end result is the exact same?

Comment (ChrisBr) [reply src/api/app/models/kiwi/image/xml_builder.rb]: Only if the version is indeed the first child. If it is on a different position, you would change the position as you add it again as first child (see line 55).

**Comment by mdeniz** [comment]
LGTM beside Chris comments

**Comment by evanrolfe** [comment]
@ChrisBr I've changed the XmlBuilder method to update preferences/version instead of remove/add. Please re-review.

**Comment by evanrolfe** [comment]
[(3) P1: Edit sections image and description](https://trello.com/c/6c513ZjA/1209-3-p1-edit-sections-image-and-description)

File: src/api/app/models/kiwi/preference.rb [hunk 3]
```diff
@@ -27,6 +27,7 @@ def editable?
 #  type_image                :integer
 #  type_containerconfig_name :string(255)
 #  type_containerconfig_tag  :string(255)
+#  version                   :string(255)
```
Comment (ChrisBr) [inline src/api/app/models/kiwi/preference.rb]: [Nitpick] Do we need a validation for the version format?

>  The image version must be a three-part version number of the format: Major.Minor.Release.

File: src/api/app/views/webui/kiwi/images/_preferences.html.haml [hunk 4]
```diff
@@ -7,11 +7,15 @@
         %strong Image type:
         %span.fill{ data: { tag: 'type_image' }}= @image.preference.type_image
       %li
-        %strong Container config name:
-        %span.fill{ data: { tag: 'type_containerconfig_name' }}= @image.preference.type_containerconfig_name
-      %li
-        %strong Container config tag:
-        %span.fill{ data: { tag: 'type_containerconfig_tag' }}= @image.preference.type_containerconfig_tag
+        %strong Version:
+        %span.fill{ data: { tag: 'version' }}= @image.preference.version
```
Comment (ChrisBr) [inline src/api/app/views/webui/kiwi/images/_preferences.html.haml]: What a about a hint text, this is from the kiwi documentation:

>  The image version must be a three-part version number of the format: Major.Minor.Release.

File: src/api/app/models/kiwi/preference.rb (hunk 3 above)
Comment (evanrolfe) [reply src/api/app/models/kiwi/preference.rb]: @adrianschroeter what do you think?

File: src/api/app/views/webui/kiwi/images/_preferences.html.haml (hunk 4 above)
Comment (evanrolfe) [reply src/api/app/views/webui/kiwi/images/_preferences.html.haml]: @ChrisBr I added some text, please re-review!

screenshot from 2017-11-27 09-30-06

**Comment by evanrolfe** [comment]
[(0) Edit sections image and description](https://trello.com/c/6c513ZjA/1209-3-p1-edit-sections-image-and-description)
